### PR TITLE
Audit the username reminder and password reset events

### PR DESCRIPTION
**Pull request by favour234**
- added "postUserActionToMetrics" function to "send username reminder" & "reset password" handlers.
- Log events for "Send username reminders" & "Reset password" in user audit history.
- added respective translations.
- fixed bug in UserAuditHistory.tsx file observed on develop branch.
- updates messages on farajaland

https://github.com/opencrvs/opencrvs-farajaland/pull/407/files

### CODE REVIEW

File: packages/user-mgnt/src/features/resstPassword/handler.ts [hunk 1]
```diff
@@ -39,6 +40,18 @@ export default async function resetPasswordSMSHandler(
...
+  const userAgent =
+    request.headers['x-real-user-agent'] || request.headers['user-agent']
+
+  await postUserActionToMetrics(
+    'SEND_PASSWORD',
+    request.headers.authorization,
+    remoteAddress,
+    userAgent
+  )
+
```
Comment (Zangetsu101) [inline packages/user-mgnt/src/features/resstPassword/handler.ts:54]: @favour234 similar to the ticket that you are working on, let's wrap the `postUserActionToMetrics` function call in a try catch block and use `logger.error` in catch block

File: packages/user-mgnt/src/features/usernameSMSReminderInvite/handler.ts [hunk 2]
```diff
@@ -35,6 +36,18 @@ export default async function usernameSMSReminderHandler(
...
+    request.headers['x-real-ip'] || request.info.remoteAddress
+  const userAgent =
+    request.headers['x-real-user-agent'] || request.headers['user-agent']
+
+  await postUserActionToMetrics(
+    'SEND_USERNAME_REMINDER',
+    request.headers.authorization,
+    remoteAddress,
+    userAgent
+  )
```
Comment (Zangetsu101) [inline packages/user-mgnt/src/features/usernameSMSReminderInvite/handler.ts:49]: Similarly here as well

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/user-mgnt/src/features/resstPassword/handler.ts (hunk 1 above)
Comment (favour234) [reply packages/user-mgnt/src/features/resstPassword/handler.ts:54]: alright..would update

File: packages/user-mgnt/src/features/usernameSMSReminderInvite/handler.ts (hunk 2 above)
Comment (favour234) [reply packages/user-mgnt/src/features/usernameSMSReminderInvite/handler.ts:49]: alright.

**Comment by Zangetsu101** [comment]
@favour234 Take a look at the second point here: https://github.com/opencrvs/opencrvs-core/pull/4279#pullrequestreview-1187660778 we need to save the `subjectPractitionerId` as a sysadmin is the one performing this action

**Comment by favour234** [comment]
> @favour234 Take a look at the second point here: [#4279 (review)](https://github.com/opencrvs/opencrvs-core/pull/4279#pullrequestreview-1187660778) we need to save the `subjectPractitionerId` as a sysadmin is the one performing this action

Updated..

File: packages/client/src/i18n/messages/views/userSetup.ts [hunk 3]
```diff
@@ -356,10 +358,23 @@ const messagesToDefine: IUserSetupMessages = {
...
   userNameReminderAuditAction: {
     defaultMessage: 'Username Reminder Requested',
     description: 'Description for User username requested audit action',
     id: 'user.profile.auditList.usernameRequested'
+  },
+
+  sendUserNameReminder: {
+    defaultMessage: 'Sent Username Reminder',
+    description: 'Description for send username reminder audit action',
+    id: 'user.profile.auditList.sentUsernameReminder'
```
Comment (euanmillar) [inline packages/client/src/i18n/messages/views/userSetup.ts:377]: @favour234 translations are required in Farajaland

**Comment by euanmillar** [comment]
@favour234 I changed the base branch to that used in ocrvs-3942-audit-v3.  I dont see why we are doing this work in 2 separate PRs are there is such a commonality of code and approach @Zangetsu101 .  When ocrvs-3942-audit-v3 merges in you will need to fix conflicts again.  So, can you address the comments and fix the tests in ocrvs-3942-audit-v3 then merge into this branch and fix conflicts please?

**Comment by favour234** [comment]
> @favour234 I changed the base branch to that used in ocrvs-3942-audit-v3.
> 
> Screenshot 2022-11-29 at 14 21 22
> 
> I dont see why we are doing this work in 2 separate PRs are there is such a commonality of code and approach @Zangetsu101 . When ocrvs-3942-audit-v3 merges in you will need to fix conflicts again. So, can you address the comments and fix the tests in ocrvs-3942-audit-v3 then merge into this branch and fix conflicts please?

alright seem there are now conflicts after the base branch was changed..would resolve .

File: packages/client/src/i18n/messages/views/userSetup.ts (hunk 3 above)
Comment (favour234) [reply packages/client/src/i18n/messages/views/userSetup.ts:381]: alright

**Pull request merged**